src/controllers/emailVault/emailVault.ts

```diff
@@ -214,7 +214,7 @@ export class EmailVaultController extends EventEmitter {
       fn && (await fn())
       this.storage.set(MAGIC_LINK_STORAGE_KEY, this.#magicLinkKeys)
       this.#requestSessionKey(email)
-    } else if (this.#shouldStopConfirmationPolling)
+    } else if (!this.#shouldStopConfirmationPolling)
```

Comment:
here we can do it again with short-circuit before ` if (ev && !ev.error) {`, because it's a bit misleading in the else if